### PR TITLE
[WIP] Fixes gh-658

### PR DESCRIPTION
**Pull request by BorzdeG**
* Fixes gh-658
* Added extension for Application
* Added tests
* Refactored tests

### CODE REVIEW

File: koin-projects/docs/reference/koin-ktor/ktor.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ fun Application.main() {
     // Install Ktor features
     install(DefaultHeaders)
     install(CallLogging)
-    install(Koin) {
```
Comment (arnaudgiuliani) [inline koin-projects/docs/reference/koin-ktor/ktor.md]: We want to keep `install(Koin)`. Can you revert this?

File: koin-projects/docs/start/getting-started/koin-for-ktor.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ Start Koin from your `Application` with the Ktor extension below:
 ```kotlin
 fun Application.main() {
     // Install Ktor features
-    install(Koin) {
```
Comment (arnaudgiuliani) [inline koin-projects/docs/start/getting-started/koin-for-ktor.md]: same, we want to keep `install(Koin)`. Can you revert this?

File: koin-projects/docs/start/getting-started/starting-koin.md [hunk 3]
```diff
@@ -76,7 +76,7 @@ Starting Koin from your `Application` extension function:
 ```kotlin
 fun Application.main() {
     // Install Ktor features
-    install(Koin) {
```
Comment (arnaudgiuliani) [inline koin-projects/docs/start/getting-started/starting-koin.md]: same, we want to keep `install(Koin)`. Can you revert this?

File: koin-projects/docs/start/quickstart/ktor.md [hunk 4]
```diff
@@ -129,7 +129,7 @@ fun Application.main() {
     install(DefaultHeaders)
     install(CallLogging)
     // Declare Koin
-    install(Koin) {
```
Comment (arnaudgiuliani) [inline koin-projects/docs/start/quickstart/ktor.md]: same, we want to keep `install(Koin)`. Can you revert this?

File: koin-projects/docs/start/setup/upgrade.md [hunk 5]
```diff
@@ -73,7 +73,7 @@ Starting Koin from your `Application` extension function:
 ```kotlin
 fun Application.main() {
     // Install Ktor features
-    install(Koin) {
```
Comment (arnaudgiuliani) [inline koin-projects/docs/start/setup/upgrade.md]: same, we want to keep `install(Koin)`. Can you revert this?

**Comment by BorzdeG** [comment]
@arnaudgiuliani , added an example

**Pull request closed without merging**